### PR TITLE
Add docker support

### PR DESCRIPTION
**Pull request by warriorzz**
[Docker](https://www.docker.com/) allows the user to containerize their applications to make it a lot easier to ship them.

You could consider an GitHub Actions workflow to build the Docker image and push it to the GitHub registry as described [here](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry). 
An example workflow would look like [this](https://gist.github.com/warriorzz/ed7ee2060b7a9892b371885c6616b31c).

### CODE REVIEW

**Comment by giorgi-o** [comment]
I've never used docker myself so I don't really know much about it. I've read the Github docs link but honestly it's all just gibberish to me. Feel free to DM me on Discord if you want to help me set it up :)

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM node:latest
```
Comment (NyCodeGHG) [inline Dockerfile:1]: the latest tag shouldn't be used here. It's better to set it to fixed version like "17-alpine". Also an alpine image reduces the image size a lot
```suggestion
FROM node:17-alpine
```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+
+WORKDIR /usr/app
+
+COPY . .
+RUN npm i
```
Comment (NyCodeGHG) [inline Dockerfile:6]:
```suggestion
COPY package.json .
RUN npm i
COPY . .
```
installing the dependencies in extra layer allows docker to cache them when they do not change

**Comment by NyCodeGHG** [comment]
@warriorzz the example is using the old docker registry. Could you update that to use the new registry ghcr.io?

**Comment by giorgi-o** [comment]
Resolved in #2

**Pull request merged**